### PR TITLE
Clean up getters and unused code

### PR DESCRIPTION
**Pull request by tinchou**
This PR is a big clean up refactor and made sense to make it all at once. Each commit removes or changes something and can be inspected individually.

Basically we remove getters and use constructor injection instead:

- `Rsk.getMessageHandler()`
- `Rsk.getPeerScoringManager()`
- `Repository.getStorageSize`
- `Repository.getStorageKeys`
- `Repository.getStorage`
- `Ethereum.getWorldManager`
- `Ethereum.callConstantFunction`
- `Ethereum.getRepository`
- `Ethereum.getPendingState`
- `Ethereum.getSnapshootTo`
- `Ethereum.getAdminInfo`
- `Ethereum.getChannelManager`
- `Ethereum.getPendingStateTransactions`
- `Ethereum.getPeerServer`
- `Ethereum.getSystemProperties`
- `WorldManager.init`
- `WorldManager.getChannelManager`
- `WorldManager.getRepository`
- `WorldManager.getNetworkStateExporter`

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 213 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 159 major
* ![MINOR][MINOR] 18 minor
* ![INFO][INFO] 28 info

#### Top 10 issues

1. ![BLOCKER][BLOCKER] [MinerServerImpl.java#L524](https://github.com/rsksmart/rskj/blob/7651b3a3390d6927419f1433483c710488ca6693/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L524): Catch Exception instead of Throwable. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1181)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L72](https://github.com/rsksmart/rskj/blob/7651b3a3390d6927419f1433483c710488ca6693/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L72): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L79](https://github.com/rsksmart/rskj/blob/7651b3a3390d6927419f1433483c710488ca6693/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L79): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L87](https://github.com/rsksmart/rskj/blob/7651b3a3390d6927419f1433483c710488ca6693/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L87): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L99](https://github.com/rsksmart/rskj/blob/7651b3a3390d6927419f1433483c710488ca6693/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L99): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L127](https://github.com/rsksmart/rskj/blob/7651b3a3390d6927419f1433483c710488ca6693/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L127): Cast one of the operands of this addition operation to a "long". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2184)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L131](https://github.com/rsksmart/rskj/blob/7651b3a3390d6927419f1433483c710488ca6693/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L131): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L155](https://github.com/rsksmart/rskj/blob/7651b3a3390d6927419f1433483c710488ca6693/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L155): Cast one of the operands of this addition operation to a "long". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2184)
1. ![MAJOR][MAJOR] [Start.java#L64](https://github.com/rsksmart/rskj/blob/7651b3a3390d6927419f1433483c710488ca6693/rskj-core/src/main/java/co/rsk/Start.java#L64): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Start.java#L82](https://github.com/rsksmart/rskj/blob/7651b3a3390d6927419f1433483c710488ca6693/rskj-core/src/main/java/co/rsk/Start.java#L82): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/core/RskFactory.java [hunk 1]
```diff
@@ -113,7 +114,9 @@ public Rsk getRsk(WorldManager worldManager,
         }
         if (rskSystemProperties.isBlocksEnabled()) {
             setupRecorder(rsk, rskSystemProperties.blocksRecorder());
-            setupPlayer(rsk, rskSystemProperties.blocksPlayer());
+            rsk.setIsPlayingBlocks(true);
```
Comment (diega) [inline rskj-core/src/main/java/co/rsk/core/RskFactory.java]: Check if moving it outside the thread doesn't change behavior.

File: rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java [hunk 2]
```diff
@@ -384,7 +387,7 @@ public String eth_gasPrice() {
 
     @Override
     public String eth_blockNumber() {
-        Blockchain blockchain = worldManager.getBlockchain();
+        Blockchain blockchain = Web3Impl.this.blockchain;
```
Comment (diega) [inline rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java]: Can't this assignment be removed? (there are a few along this class)

File: rskj-core/src/main/java/co/rsk/core/RskFactory.java (hunk 1 above)
Comment (tinchou) [reply rskj-core/src/main/java/co/rsk/core/RskFactory.java]: Great finding!

File: rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java (hunk 2 above)
Comment (tinchou) [reply rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java]: For sure!

**Comment by rskops** [comment]
SonarQube analysis reported 209 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 155 major
* ![MINOR][MINOR] 18 minor
* ![INFO][INFO] 28 info

#### Top 10 issues

1. ![BLOCKER][BLOCKER] [MinerServerImpl.java#L524](https://github.com/rsksmart/rskj/blob/45cec711095865bccfea38033ca2ed66db8b0a48/rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java#L524): Catch Exception instead of Throwable. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1181)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L72](https://github.com/rsksmart/rskj/blob/45cec711095865bccfea38033ca2ed66db8b0a48/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L72): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L79](https://github.com/rsksmart/rskj/blob/45cec711095865bccfea38033ca2ed66db8b0a48/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L79): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L87](https://github.com/rsksmart/rskj/blob/45cec711095865bccfea38033ca2ed66db8b0a48/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L87): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L99](https://github.com/rsksmart/rskj/blob/45cec711095865bccfea38033ca2ed66db8b0a48/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L99): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L127](https://github.com/rsksmart/rskj/blob/45cec711095865bccfea38033ca2ed66db8b0a48/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L127): Cast one of the operands of this addition operation to a "long". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2184)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L131](https://github.com/rsksmart/rskj/blob/45cec711095865bccfea38033ca2ed66db8b0a48/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L131): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [TxBuilderEx.java#L155](https://github.com/rsksmart/rskj/blob/45cec711095865bccfea38033ca2ed66db8b0a48/rskj-core/src/main/java/co/rsk/mine/TxBuilderEx.java#L155): Cast one of the operands of this addition operation to a "long". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2184)
1. ![MAJOR][MAJOR] [Start.java#L65](https://github.com/rsksmart/rskj/blob/45cec711095865bccfea38033ca2ed66db8b0a48/rskj-core/src/main/java/co/rsk/Start.java#L65): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Start.java#L83](https://github.com/rsksmart/rskj/blob/45cec711095865bccfea38033ca2ed66db8b0a48/rskj-core/src/main/java/co/rsk/Start.java#L83): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**